src/app/pages/questions/services/questions.service.ts

```diff
@@ -72,29 +76,13 @@ export class QuestionsService {
     }
   }
 
-  showESMSleepQuestion() {
-    // Note: First ESM will show sleep question
-    return new Date().getHours() <= 9
-  }
-
-  showESMRatingQuestion() {
-    // Note: Last ESM will show rating question
-    // TODO: Fix hardcoded values
-    return new Date().getHours() >= 19
-  }
-
-  isESM(title) {
-    return title === 'ESM'
-  }
-
   processQuestions(title, questions: any[]) {
-    if (this.isESM(title)) {
-      const length = questions.length
-      const first = this.showESMSleepQuestion() ? 0 : 1
-      const last = this.showESMRatingQuestion() ? length - 1 : length - 2
-      return questions.slice(first, last)
-    }
-    return questions
+    return this.getHiddenQuestions().then(res => {
+      if (!res[title]) return questions
+      const questionsToHide = res[title].split(',').map(i => Number(i))
```

Comment:
What happens if the format is not right. For example, Someone put a Word instead of a number. I think we should catch any exceptions and log an error event and then just proceed with all the questions. So as to not break the app

Reply:
Okay made the changes.